test.disabled/ejabberd_tests/tests/push_integration_SUITE.erl

```diff
@@ -0,0 +1,553 @@
+                PushPortStr = getenv("MONGOOSE_PUSH_PORT", ?DEFAULT_MONGOOSE_PUSH_PORT),
+                {ok, _} = gen_tcp:connect("localhost", list_to_integer(PushPortStr), []),
+                escalus:init_per_suite(Config)
+            catch
+                _:_ ->
+                    {skip,  "MongoosePush is not available"}
+            end;
+        false ->
+            {skip,  "This test suite requires Erlang's SSL 7.0+ that is available in"
+                    "Erlang/OTP 18.0+"}
+    end.
+
+
+end_per_suite(Config) ->
+    escalus_fresh:clean(),
+    rpc(mongoose_http_client, stop_pool, [mongoose_push_http]),
+    rpc(mongoose_http_client, stop, []),
+    dynamic_modules:restore_modules(domain(), Config),
+    escalus:end_per_suite(Config).
+
+init_per_group(_, Config) ->
+    %% Some cleaning up
+    FCM = connect_to(fcm),
+    APNS = connect_to(apns),
+    {ok, 200, _} = h2_req(FCM, post, <<"/reset">>),
+    {ok, 200, _} = h2_req(APNS, post, <<"/reset">>),
+    rpc(mod_muc_light_db_backend, force_clear, []),
+
+    Config.
+
+end_per_group(_, Config) ->
+    Config.
+
+init_per_testcase(CaseName, Config0) ->
+    Config1 = escalus_fresh:create_users(Config0, [{bob, 1}, {alice, 1}, {kate, 1}]),
+    Config = [{case_name, CaseName} | Config1],
+    escalus:init_per_testcase(CaseName, Config).
+
+end_per_testcase(CaseName, Config) ->
+    escalus:end_per_testcase(CaseName, Config).
+
+%%--------------------------------------------------------------------
+%% GROUP pm_msg_notifications
+%%--------------------------------------------------------------------
+
+pm_msg_notify_on_apns_no_click_action(Config) ->
+    escalus:story(
+        Config, [{bob, 1}, {alice, 1}],
+        fun(Bob, Alice) ->
+            PubsubJID = <<"pubsub.localhost">>,
+            Node = {_, NodeName} = pubsub_node(),
+            AliceJID = bare_jid(Alice),
+            DeviceToken = gen_token(),
+
+            pubsub_tools:create_node(Bob, Node, [{type, <<"push">>}]),
+            escalus:send(Bob, enable_stanza(PubsubJID, NodeName,
+                                            [{<<"service">>, <<"apns">>},
+                                             {<<"device_id">>, DeviceToken}])),
+            escalus:assert(is_result, escalus:wait_for_stanza(Bob)),
+            become_unavailable(Bob),
+
+            escalus:send(Alice, escalus_stanza:chat_to(Bob, <<"OH, HAI!">>)),
+
+            [Notification = #{}] = get_push_logs(apns, DeviceToken, Config),
+
+            APNSData = maps:get(<<"aps">>, maps:get(<<"request_data">>, Notification)),
+            APNSAlert = maps:get(<<"alert">>, APNSData),
+            ?assertMatch(#{<<"body">> := <<"OH, HAI!">>}, APNSAlert),
+            ?assertMatch(#{<<"title">> := AliceJID}, APNSAlert),
+            ?assertMatch(#{<<"badge">> := 1}, APNSData),
+            ?assertMatch(#{<<"category">> := null}, APNSData),
+
+            ok
+        end).
+
+pm_msg_notify_on_fcm_no_click_action(Config) ->
```

Comment:
Looks like there is quite some common lines between this test case and the previous one. I strongly recommend eliminating duplicated code (even in tests).